### PR TITLE
#1209 fix failOnError for OptimizeMojo

### PR DESCRIPTION
**Pull request by mximp**
#1209 fix

### CODE REVIEW

File: eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java [hunk 1]
```diff
@@ -158,6 +161,25 @@ private XML optimize(final Path file) throws FileNotFoundException {
                 new Home().rel(dir)
             );
         }
+        if (this.failOnError) {
+            trn = trn.with(
+                new StLambda(
+                    xml -> {
+                        final List<String> errors = xml.xpath("//error[@severity='error']/text()");
```
Comment (yegor256) [inline eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java:168]: @mximp it feels that this code is duplicated. We have this check somewhere else, right? Maybe it's better to make an XSL that does this validation and throws `<xsl:message>`. Then, we can simply add such XSL to the train and that's it.

**Comment by yegor256** [comment]
@mximp check my comment

File: eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java (hunk 1 above)
Comment (mximp) [reply eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java:168]: @yegor256 correct: similar check is done in `Sanitized` which called from `TranspileMojo`. It analyses an XMIR from file.
I didn't quite get your idea with another xsl: how would we address `failOnError` flag from within XSL? Passing XSL params via `XSL::with()` method would require almost as much code as we have now to set up new XSL with parameter and passing it.

Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java:168]: @mximp we just create an XSL stylesheet, which finds `error[@severity=error]` and terminates processing with `<xsl:message>`. Then, we add this XSL to the train with one `.with()` command.

**Comment by mximp** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@mximp let's use this XSL in `TranspileMojo` too?

**Comment by mximp** [comment]
> @mximp let's use this XSL in `TranspileMojo` too?

@yegor256 `TranspileMojo` doesn't produce any errors within `XMIR`. All errors are added during parsing and optimization. So what is the point?

**Comment by yegor256** [comment]
@mximp the class `Sanitized` is doing almost exactly the same as the XSL you just created. Maybe we can remove `Sanitized` and only use the XSL?

**Comment by mximp** [comment]
@yegor256 I guess it's enough to have this validation by XSL in `OptimizeMojo`. And we can remove the check entirely from `TranspileMojo`. WDYT?

**Comment by yegor256** [comment]
@mximp good idea

**Comment by mximp** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@mximp now it's perfect!

**Pull request merged**